File: 14. Identity Matrix Check.cpp
// Topic: Identity Matrix Check
#include <bits/stdc++.h>
using namespace std;
int main()
{
    vector<vector<int>> A = {{1, 0, 0}, {0, 1, 0}, {0, 0, 1}};
    int n = A.size();
    bool ok = true;
    for (int i = 0; i < n && ok; i++)
        for (int j = 0; j < n; j++)
        {
            if (i == j)
            {
                if (A[i][j] != 1)
                {
                    ok = false;
                    break;
                }
            }
            else if (A[i][j] != 0)
            {
                ok = false;
                break;
            }
        }
    cout << (ok ? "Identity\n" : "Not Identity\n");
}

// TC: O(n^2)
// SC: O(1)

File: 14. Identity Matrix Check.py
"""Topic: Identity Matrix Check"""
def is_identity(A):
    n=len(A)
    for i in range(n):
        for j in range(n):
            if i==j and A[i][j]!=1: return False
            if i!=j and A[i][j]!=0: return False
    return True

if __name__=='__main__':
    A=[[1,0,0],[0,1,0],[0,0,1]]
    print(is_identity(A))

# TC: O(n^2)
# SC: O(1)
